clientv3/watch.go

```diff
@@ -614,13 +603,34 @@ func (w *watchGrpcStream) serveSubstream(ws *watcherStream, resumec chan struct{
 				// shutdown from closeSubstream
 				return
 			}
-			// TODO pause channel if buffer gets too large
-			ws.buf = append(ws.buf, wr)
+
+			if wr.Created {
+				select {
+				case ws.initReq.retc <- ws.outc:
+					// send first creation event only if requested
+					if ws.initReq.createdNotify {
+						ws.outc <- *wr
```

Comment:
@heyitsanthony  This still sends the second create event in the new test, so it fails.

Should we add a var to record if it's second event or not?
Any suggestion? Thanks.